### ISSUE TITLE
[Android] PlatformException(error, Serializer for class 'Any' is not found.

### ISSUE THREAD

**Issue by bladeofgod** [781flyingdutchman/background_downloader#383]
**Describe the bug**
Hi, i got this native-exception on android platform, ios is just fine.
When i randomly download video(big size maybe) with "loading dialog",  i will stuck(loading dialog will never dismiss) even i handled `FileDownloader().download(...)` 's `result.exception`, because the exception is passed by zone channel.

BTW, awsome plugin !

1. randomly download some video(big size is better)
2. will see exception at terminal

**Expected behavior**
fixed it or get exception from plugin channel, not in platform expcetion.
thanks :)

**Logs**
```
exception : PlatformException(error, Serializer for class 'Any' is not found.
Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
, null, ed.f: Serializer for class 'Any' is not found.
Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.

	at id.u0.e(Unknown Source:11)
	at ed.i.a(Unknown Source:19)
	at ed.h.a(Unknown Source:0)
	at com.bbflight.background_downloader.a.P0(Unknown Source:141)
	at com.bbflight.background_downloader.a.Z(Unknown Source:0)
	at com.bbflight.background_downloader.a$o.q(Unknown Source:146)
	at bc.a.h(Unknown Source:11)
	at uc.u0.run(Unknown Source:93)
	at uc.c1.v0(Unknown Source:79)
	at uc.e.b1(Unknown Source:23)
	at uc.h.a(Unknown Source:73)
	at uc.g.c(Unknown Source:0)
	at uc.h.b(Unknown Source:6)
	at uc.g.d(Unknown Source:0)
	at com.bbflight.background_downloader.a.onMethodCall(Unknown Source:17)
	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(Unknown Source:17)
	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(Unknown Source:18)
	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0(Unknown Source:41)
	at io.flutter.embedding.engine.dart.DartMessenger.a(Unknown Source:0)
	at io.flutter.embedding.engine.dart.a.run(Unknown Source:12)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:257)
	at android.os.Looper.loop(Looper.java:368)
	at android.app.ActivityThread.main(ActivityThread.java:8839)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
```

**Comment by 781flyingdutchman** [comment]
Thanks for raising the issue. These logs are obfuscated (using pro guard) and therefore hard to interpret. Can you either de-obfuscate it (using the file that ProGuard generated) or replicate it in debug mode without ProGuard?

Also, can you confirm you have upgraded to the latest version of the plugin?

**Comment by bladeofgod** [comment]
> Thanks for raising the issue. These logs are obfuscated (using pro guard) and therefore hard to interpret. Can you either de-obfuscate it (using the file that ProGuard generated) or replicate it in debug mode without ProGuard?
> 
> Also, can you confirm you have upgraded to the latest version of the plugin?

thanks for your reply. 
my plugin version is ^8.5.6, and the log is shown below:

```
/flutter (30659): err_msg : exception : PlatformException(error, Serializer for class 'Any' is not found.
I/flutter (30659): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (30659): , null, kotlinx.serialization.SerializationException: Serializer for class 'Any' is not found.
I/flutter (30659): Please ensure that class is marked as '@Serializable' and that the serialization compiler plugin is applied.
I/flutter (30659): 
I/flutter (30659): 	at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
I/flutter (30659): 	at kotlinx.serialization.SerializersKt__SerializersKt.noCompiledSerializer(Serializers.kt:366)
I/flutter (30659): 	at kotlinx.serialization.SerializersKt.noCompiledSerializer(Unknown Source:1)
I/flutter (30659): 	at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
I/flutter (30659): 	at com.bbflight.background_downloader.BDPlugin.access$methodUpdateChunkStatus(BDPlugin.kt:56)
I/flutter (30659): 	at com.bbflight.background_downloader.BDPlugin$onMethodCall$1.invokeSuspend(BDPlugin.kt:392)
I/flutter (30659): 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
I/flutter (30659): 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
I/flutter (30659): 	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:280)
I/flutter (30659): 	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
I/flutter (30659): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
I/flutter (30659): 	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
I/flutter (30659): 	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
I/flutter (30659): 	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
I/flutter (30659): 	at com.bbflight.background_downloader.BDPlugin.onMethodCall(BDPlugin.kt:376)
I/flutter (30659): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:267)
I/flutter (30659): 	at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:292)
I/flutter (30659): 	at io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0(DartMessenger.java:319)
I/flutter (30659): 	at io.flutter.embedding.engine.dart.DartMessenger.$r8$lambda$2j2MERcK825A5j1fv5sZ7xB2Iuo(Unknown Source:0)
I/flutter (30659): 	at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
I/flutter (30659): 	at android.os.Handler.handleCallback(Handler.java:958)
I/flutter (30659): 	at android.os.Handler.dispatchMessage(Handler.java:99)
I/flutter (30659): 	at android.os.Looper.loopOnce(Looper.java:257)
I/flutter (30659): 	at android.os.Looper.loop(Looper.java:368)
I/flutter (30659): 	at android.app.ActivityThread.main(ActivityThread.java:8839)
I/flutter (30659): 	at java.lang.reflect.Method.invoke(Native Method)
I/flutter (30659): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:572)
I/flutter (30659): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
I/flutter (30659): )  , stackTrace : #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:648:7)
I/flutter (30659): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:334:18)
I/flutter (30659): <asynchronous suspension>
```

**Comment by 781flyingdutchman** [comment]
I can't match those logs with version 8.5.6 - I suspect they're from an earlier version. Can you reproduce the bug with 8.5.6?

**Comment by bladeofgod** [comment]
> I can't match those logs with version 8.5.6 - I suspect they're from an earlier version. Can you reproduce the bug with 8.5.6?

yes， the log is from 8.5.6 , and i checked plugin version in `pubspec.lock` file :
```
  background_downloader:
    dependency: "direct main"
    description:
      name: background_downloader
      sha256: "6b73fa5d20c47e855f6ef3ed6fb3e0d164141d8ae7d43ca0a42c78f90eaa15e7"
      url: "https://pub.flutter-io.cn"
    source: hosted
    version: "8.5.6"
```
And,  /app/build.gradle info like this :  
```
plugins {
    id "com.android.application"
    id "kotlin-android"
    id "dev.flutter.flutter-gradle-plugin"
    id 'org.jetbrains.kotlin.plugin.serialization' version '1.9.20'
}
```

**Comment by 781flyingdutchman** [comment]
There is somethingoff though: the line where the log says the error occurs is `at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96)
` and that line is a declaration `val remainingBytesToDownload = mutableMapOf<String, Long>()` so definitely not where the error actually occurs.  Please reproduce the error, share the code that reproduices. it and share the logs from those reproducable issues.

**Comment by bladeofgod** [comment]
> There is somethingoff though: the line where the log says the error occurs is `at com.bbflight.background_downloader.BDPlugin.methodUpdateChunkStatus(BDPlugin.kt:96) ` and that line is a declaration `val remainingBytesToDownload = mutableMapOf<String, Long>()` so definitely not where the error actually occurs. Please reproduce the error, share the code that reproduices. it and share the logs from those reproducable issues.

sorry  mate, I'm in vacation last few days, and I will provide code after next day.
And , in addition , if this error is occured by server or connection ,  I think maybe just catch it and confirm this chunk-download-task failed , than do retry or  post to flutter.

**Comment by 781flyingdutchman** [comment]
You've got to match up the logs with the version you are using. Everything looks like you are using an older version that had this issue. I'm going to close this issue for now.

**Comment by bladeofgod** [comment]
> You've got to match up the logs with the version you are using. Everything looks like you are using an older version that had this issue. I'm going to close this issue for now.

ok, i will looking for it (maybe kotlin version i think), thanks.

**Comment by 781flyingdutchman** [comment]
Fixed in 8.7.0
